python/cudf/cudf/core/column/decimal.py

```diff
@@ -20,12 +19,11 @@ def from_arrow(cls, data: pa.Array):
             if mask_buf is None
             else pa_mask_buffer_to_mask(mask_buf, len(data))
         )
-        data_128 = cp.array(np.frombuffer(data.buffers()[1]).view("uint8"))
-        data_64 = cp.empty(data_128.size // 2, dtype=data_128.dtype)
-        alternating_chunks.forall(data_64.size)(data_128, data_64, 4)
+        data_128 = cp.array(np.frombuffer(data.buffers()[1]).view("int64"))
+        data_64 = data_128[::2].copy()
```

Comment:
It would be really nice if we could at least warn or throw when we're losing precision on the data. Especially situations where you have a larger scale so the significant data is presumably held in the first 64 bits of the 128 bit element.

Reply:
One idea is to throw when `precision >= floor(log10(INT64_MAX))`, i.e., `precision >= 18`.

Reply:
I'd support this as well even if it requires a scan.

Reply:
For Arrow this matters since we can just make direct memory copies for the most part. I'd really like to avoid data introspection in the case of Arrow.

Reply:
Done -- we throw when constructing a `DecimalDtype` with `precision>18`, or setting its precision to a value `>18`.